flopy/utils/mfreadnam.py

```diff
@@ -153,32 +152,29 @@ def parsenamefile(namfilename, packages, verbose=True):
 
         # need make filenames with paths system agnostic
         if "/" in fpath:
-            raw = fpath.split("/")
+            fpath = PurePosixPath(fpath)
```

Comment:
If I constructed a path with "/" on windows it would assume it is not on Windows OS even if that was not the case. Would a specific OS check be better?

Reply:
Both PurePosixPath and PureWindowsPath behave the same on any OS, so no OS checks are needed for this statement. Also abominations like `PureWindowsPath(r"../prev\next/file.txt").parts` correctly pick out `('..', 'prev', 'next', 'file.txt')`. I've swapped the logic around so paths with `\` are processed by PureWindowsPath first.